### PR TITLE
added custom analyzer structs

### PR DESCRIPTION
**Pull request by abner-dou**
## Description

Added structs to manage custom analyzers in index struct, also documentation was added to code

Link to any related issue(s): 
INTMDB-17
## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: mongodbatlas/search.go [hunk 1]
```diff
@@ -280,7 +280,7 @@ type IndexField struct {
 	IndexShapes    *bool                  `json:"indexShapes,omitempty"`
 }
 
-// SearchAnalyzer custom analyzer definition.
+// SearchAnalyzer search analyzer definition.
```
Comment (gssbzn) [inline mongodbatlas/search.go:283]: if updating this be mindful to write it like simple sentence so this page can be read
https://pkg.go.dev/go.mongodb.org/atlas@v0.8.0/mongodbatlas#SearchAnalyzer

**Review by gssbzn — APPROVED** [review]
LGTM just be mindful on the godocs as they should be written as a sentence

**Review by themantissa — COMMENTED** [review]
LGTM!

**Review by themantissa — APPROVED** [review]
LGTM, sorry hit comment first time!

**Review by thetonymaster — APPROVED** [review]

**Pull request merged**